okhttp/src/main/java/okhttp3/internal/http/HttpEngine.java

```diff
@@ -940,6 +941,17 @@ public Request followUpRequest() throws IOException {
 
         return requestBuilder.url(url).build();
 
+      case HTTP_CLIENT_TIMEOUT:
+        // 408's are rare in practice, but some servers like HAProxy use this response code. The
+        // spec says that we may repeat the request without modifications. Modern browsers also
+        // repeat the request (even non-idempotent ones.)
+        boolean retryableBody = requestBodyOut == null || requestBodyOut instanceof RetryableSink;
+        if (callerWritesRequestBody && !retryableBody) {
+          return null;
+        }
+
+        return userRequest.newBuilder().build();
```

Comment:
Is a copy needed here? You can't just return the original `userRequest` directly?